ivy/functional/ivy/losses.py

```diff
@@ -383,3 +383,56 @@ def sparse_cross_entropy(
     return ivy.cross_entropy(
         true, pred, axis=axis, epsilon=epsilon, reduction=reduction, out=out
     )
+
+
+@handle_exceptions
+@handle_nestable
+@handle_array_like_without_promotion
+@inputs_to_ivy_arrays
+@handle_array_function
+def huber_loss(
+    y_true: Union[ivy.Array, ivy.NativeArray],
+    y_pred: Union[ivy.Array, ivy.NativeArray],
+    /,
+    delta: float = 1.0,
+    out: Optional[ivy.Array] = None,
+) -> ivy.Array:
+    """
+    Compute the Huber loss (smooth L1 loss) between true and predicted values.
+
+    Parameters
+    ----------
+    y_true : array_like
+        The true (ground truth) values.
+    y_pred : array_like
+        The predicted values by the model.
+    delta : float, optional
+        The threshold parameter that determines the point where the loss transitions from
+        squared error to absolute error. Default is 1.0.
+    out : array_like, optional
+        Optional output array, for writing the result to. It must have a shape
+        that the inputs broadcast to.
+
+    Returns
+    -------
+    ret : array_like
+        The Huber loss between the true and predicted values.
+
+    Examples
+    --------
+    >>> y_true = ivy.array([2, 4, 7, 1])
+    >>> y_pred = ivy.array([2.5, 3.5, 8, 0.8])
+    >>> huber_loss(y_true, y_pred, delta=1.0)
+    ivy.array([0.125, 0.125, 0.5  , 0.125])
+
+    >>> huber_loss(y_true, y_pred, delta=2.0)
+    ivy.array([0.125, 0.125, 0.5  , 0.2  ])
+
+    >>> huber_loss(y_true, y_pred, delta=0.5)
+    ivy.array([0.25 , 0.25 , 0.   , 0.125])
+    """
+    ivy.utils.assertions.assert_float_param(delta, 'delta', allow_inf=False)
```

Comment:
There's no function `assert_float_param` in `assertions`. I am not sure what exactly you are trying to do here.  Could you please explain this?